### PR TITLE
Drop EL7 from mash for nightly

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: koji/collection-mash-split.py [hunk 1]
```diff
@@ -400,12 +400,15 @@ def main():
...
         CONFIG["comps_path"] = "/mnt/koji/mash/comps/foreman"
 
-        mashes = [MashConfig(collection, version, "rhel7-dist", "rhel7", "RHEL/7")]
         if collection == 'foreman' and version not in ('3.1', '3.0', '2.5'):
             modulemd_suffix = 'el8'
         else:
             modulemd_suffix = None
-        mashes.append(MashConfig(collection, version, "el8", "el8", "RHEL/8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context))
+
+        mashes = MashConfig(collection, version, "el8", "el8", "RHEL/8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context)
```
Comment (ekohl) [inline koji/collection-mash-split.py:408]: Should be a list:
```suggestion
        mashes = [MashConfig(collection, version, "el8", "el8", "RHEL/8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context)]
```

File: koji/collection-mash-split.py [hunk 2]
```diff
@@ -400,12 +400,15 @@ def main():
...
+
+        if LooseVersion(version) <= LooseVersion("3.3"):
+            mashes.append([MashConfig(collection, version, "rhel7-dist", "rhel7", "RHEL/7")])
```
Comment (ekohl) [inline koji/collection-mash-split.py:411]: And this shouldn't be a list:
```suggestion
            mashes.append(MashConfig(collection, version, "rhel7-dist", "rhel7", "RHEL/7"))
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: koji/collection-mash-split.py [hunk 3]
```diff
@@ -400,12 +400,15 @@ def main():
...
         CONFIG["comps_path"] = "/mnt/koji/mash/comps/foreman"
 
-        mashes = [MashConfig(collection, version, "rhel7-dist", "rhel7", "RHEL/7")]
         if collection == 'foreman' and version not in ('3.1', '3.0', '2.5'):
             modulemd_suffix = 'el8'
         else:
             modulemd_suffix = None
-        mashes.append(MashConfig(collection, version, "el8", "el8", "RHEL/8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context))
+
+        mashes = [MashConfig(collection, version, "el8", "el8", "RHEL/8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context)]
```
Comment (ekohl) [inline koji/collection-mash-split.py:408]: It's my (longer term) desire to align things and I'd like to change `RHEL/8` to `el8`. I wonder if now is a good moment. We'd have to change forklift to stop using the older repos, but on the other hand it'd ensure you're not using stale EL7 content.

**Review by ekohl — APPROVED** [review]
Should we purge EL 7 nightly from http://koji.katello.org/releases/

File: koji/collection-mash-split.py [hunk 4]
```diff
@@ -405,7 +405,7 @@ def main():
         else:
             modulemd_suffix = None
 
-        mashes = [MashConfig(collection, version, "el8", "el8", "RHEL/8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context)]
+        mashes = [MashConfig(collection, version, "el8", "el8", "el8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context)]
```
Comment (ekohl) [inline koji/collection-mash-split.py:408]: If you change this, you should probably also change EL7 to match and then our forklift config too:
https://github.com/theforeman/forklift/blob/fc38b2ecaf82cc8ee8901056308fab846fa7368a/roles/foreman_repositories/tasks/redhat_staging_repos.yml#L13
https://github.com/theforeman/forklift/blob/fc38b2ecaf82cc8ee8901056308fab846fa7368a/roles/foreman_repositories/tasks/redhat_staging_repos.yml#L24

Comment (ehelms) [reply koji/collection-mash-split.py:408]: Yea, good call, let's just do it all.

**Comment by ehelms** [comment]
These two follow on changes are needed:

 - https://github.com/theforeman/forklift/pull/1520
 - https://github.com/theforeman/jenkins-jobs/pull/234

File: koji/collection-mash-split.py [hunk 5]
```diff
@@ -452,12 +448,13 @@ def main():
...
 
         mashes = [
-            MashConfig(collection, version, "rhel7", "server-rhel7", "katello/el7"),
             MashConfig(collection, version, "el8", "el8", "katello/el8", modulemd_suffix=modulemd_suffix, modulemd_version=modulemd_version, modulemd_context=modulemd_context),
-            mash_config_candlepin,
             el8_candlepin,
         ]
 
+        if LooseVersion(version) <= LooseVersion("4.5"):
+            mahses.append(MashConfig(collection, version, "rhel7", "server-rhel7", "katello/el7"))
```
Comment (ekohl) [inline koji/collection-mash-split.py:456]:
```suggestion
            mashes.append(MashConfig(collection, version, "rhel7", "server-rhel7", "katello/el7"))
```

File: koji/collection-mash-split.py [hunk 6]
```diff
@@ -452,12 +448,13 @@ def main():
...
+            mashes.append(MashConfig(collection, version, "thirdparty-candlepin-rhel7", "candlepin-server-rhel7", "candlepin/el7")
```
Comment (ekohl) [inline koji/collection-mash-split.py:457]:
```suggestion
            mashes.append(MashConfig(collection, version, "thirdparty-candlepin-rhel7", "candlepin-server-rhel7", "candlepin/el7"))
```

**Pull request merged**